plugins/javadoc/src/main/kotlin/javadoc/renderer/KorteJavadocRenderer.kt

```diff
@@ -0,0 +1,186 @@
+        is TreeViewPage -> "treePage.korte"
+        else -> ""
+    }
+
+    private fun CoroutineScope.renderNode(node: PageNode, path: String = "") {
+        if (node is JavadocPageNode) {
+            renderJavadocPageNode(node)
+        } else if (node is RendererSpecificPage) {
+            renderSpecificPage(node, path)
+        }
+    }
+
+    private fun CoroutineScope.renderModulePageNode(node: JavadocModulePageNode) {
+        val link = "."
+        val name = "index"
+        val pathToRoot = ""
+
+        val contentMap = JavadocContentToTemplateMapTranslator(locationProvider, context).templateMapForPageNode(node, pathToRoot)
+
+        writeFromTemplate(outputWriter, "$link/$name".toNormalized(), "tabPage.korte", contentMap.toList())
+        node.children.forEach { renderNode(it, link) }
+    }
+
+    private fun CoroutineScope.renderJavadocPageNode(node: JavadocPageNode) {
+        val link = locationProvider.resolve(node, skipExtension = true)
+        val dir = Paths.get(link).parent?.let { it.toNormalized() }.orEmpty()
+        val pathToRoot = dir.split("/").filter { it.isNotEmpty() }.joinToString("/") { ".." }.let {
+            if (it.isNotEmpty()) "$it/" else it
+        }
+
+        val contentMap = JavadocContentToTemplateMapTranslator(locationProvider, context).templateMapForPageNode(node, pathToRoot)
+        writeFromTemplate(outputWriter, link, templateForNode(node), contentMap.toList())
+        node.children.forEach { renderNode(it, link.toNormalized()) }
+    }
+
+    fun CoroutineScope.renderSpecificPage(node: RendererSpecificPage, path: String) = launch {
+        when (val strategy = node.strategy) {
+            is RenderingStrategy.Copy -> outputWriter.writeResources(strategy.from, "")
+            is RenderingStrategy.Write -> outputWriter.writeHtml(path, strategy.text)
+            is RenderingStrategy.Callback -> outputWriter.writeResources(
+                path,
+                strategy.instructions(this@KorteJavadocRenderer, node)
+            )
+            RenderingStrategy.DoNothing -> Unit
+        }
+    }
+
+    fun Pair<String, String>.pairToTag() =
+        """<th class="colFirst" scope="row">${first}</th>\n<td class="colLast">${second}</td>"""
+
+    fun DRI.toLink(context: PageNode? = null) = locationProvider.resolve(this, emptySet(), context)
+
+    private fun Path.toNormalized() = this.normalize().toFile().toString()
+    private fun String.toNormalized() = Paths.get(this).toNormalized()
+
+    private suspend fun OutputWriter.writeHtml(path: String, text: String) = write(path, text, ".html")
+    private fun CoroutineScope.writeFromTemplate(
+        writer: OutputWriter,
+        path: String,
+        template: String,
+        args: List<Pair<String, *>>
+    ) = launch {
+        val tmp = templateRenderer.render(template, *(args.toTypedArray()))
+        writer.writeHtml(path, tmp)
+    }
+
+    fun getTemplateConfig() = TemplateConfig().also { config ->
```

Comment:
Public?

Reply:
I think this file has several functions where it is not clear whether or not you really want to have those public 🤔

Reply:
Ok, privatization was done.